### PR TITLE
Update buildall script and the build doc [skip ci]

### LINKED ISSUE #4016: [DOC] Document latest build improvements
## Report incorrect documentation

After recent build updates when building locally we no longer need to install aggregator artifacts and we gained the ability to build arbitrary shim mixes in the dist jar without having to rely on the hardcoded profiles 

**Location of incorrect documentation**
https://github.com/NVIDIA/spark-rapids/blob/92cc33489b3c88e3d787cf8182ea1660646e3c17/CONTRIBUTING.md#L53

### PR DESCRIPTION
**Pull request by gerashegalov**
- update doc
- update buildall

Closes #4016 

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

File: build/buildall [hunk 1]
```diff
@@ -25,16 +25,19 @@ while [[ "$1" != "" ]] ; do
...
-  echo "   --module=MODULE          after finishing parallel builds, resume from dist and build"
-  echo "                            up to and including module MODULE, e.g., integration_tests"
-  echo "   --parallel=N             build in parallel, N is passed via -P to xargs. NOTE: standard maven does not support"
-  echo "                            concurrent installs. If you see an error related to a maven-metadata-local.xml in"
-  echo "                            your local m2 cache, delete its parent directory and retry."
+  echo "   -h, --help"
+  echo "        print this help message"
+  echo "   -p=DIST_PROFILE, --profile=DIST_PROFILE"
+  echo "        use this profile for the dist module, default: noSnapshots, also supported: snapshots, minimumFeatureVersionMix,"
+  echo "        snapshotsWithDataBricks, and noSnapshotsWithDataBricks. NOTE: the Databricks-related spark3XYdb shims"
```
Comment (jlowe) [inline build/buildall:32]:
```suggestion
  echo "        snapshotsWithDatabricks, and noSnapshotsWithDatabricks. NOTE: the Databricks-related spark3XYdb shims"
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -50,31 +50,34 @@ You can find all available build versions in the top level pom.xml file. If you
...
 
 By default the distribution jar only includes code for a single version of Spark. If you want
-to create a jar with multiple versions we currently have 4 options.
+to create a jar with multiple versions we have the following options.
 
 1. Build for all Apache Spark versions and CDH with no SNAPSHOT versions of Spark, only released. Use `-PnoSnapshots`.
 2. Build for all Apache Spark versions and CDH including SNAPSHOT versions of Spark we have supported for. Use `-Psnapshots`.
 3. Build for all Apache Spark versions, CDH and Databricks with no SNAPSHOT versions of Spark, only released. Use `-PnoSnaphsotsWithDatabricks`.
 4. Build for all Apache Spark versions, CDH and Databricks including SNAPSHOT versions of Spark we have supported for. Use `-PsnapshotsWithDatabricks`
+5. Build for an arbitrary combination of Apache Spark versions using `-Dincluded_buildvers=<CSV list of build versions>` e.g., `-Dincluded_buildvers=312,330`
```
Comment (tgravescs) [inline CONTRIBUTING.md:66]: might be more generic to say comma separated list of build versions

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -84,9 +87,9 @@ There is a build script `build/buildall` that automates the local build process.
 
 By default, it builds everything that is needed to create a distribution jar for all released (noSnapshots) Spark versions except for Databricks. Other profiles that you can pass using `--profile=<distribution profile>` include
 - `snapshots`
-- `minimumFeatureVersionMix` that currently includes 302, 311cdh, 312, 320 is recommended for catching incompatibilites already in the local development cycle
+- `minimumFeatureVersionMix` that currently includes 302, 311cdh, 312, 320 is recommended for catching incompatibilities already in the local development cycle
 
-For initial quick iterations we can use `--profile=<buildver>` to build a single-shim version. e.g., `-Dbuildver=320` for Spark 3.2.0
+For initial quick iterations we can use `--profile=<buildver>` to build a single-shim version. e.g., `--profile=301,321` for Spark 3.0.1 and 3.2.1
```
Comment (tgravescs) [inline CONTRIBUTING.md:92]: text says single shim version but example shows multiple, I think we should stick with 1  in the example section.

File: build/buildall [hunk 4]
```diff
@@ -25,16 +25,19 @@ while [[ "$1" != "" ]] ; do
...
+  echo "        are not built locally, the jars are fetched prebuilt from a remote Maven repo."
+  echo "        You can also supply a comma-separated list of build versions. E.g., --profile=320,330 will build only"
+  echo "        the distribution jar only for 3.2.0 and 3.3.0"
+  echo "   -m=MODULE, --module=MODULE"
+  echo "        after finishing parallel builds, resume from dist and build up to and including module MODULE."
+  echo "        E.g., --module=integration_tests"
+  echo "   -P=N, --parallel=N"
+  echo "        Build in parallel, N is passed via -P to xargs."
```
Comment (tgravescs) [inline build/buildall:40]: add default is 4

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Review by tgravescs — APPROVED** [review]

**Pull request merged**